addons/mass_mailing/models/mailing.py

```diff
@@ -260,15 +260,34 @@ def _compute_statistics(self):
             self.browse(row.pop('mailing_id')).update(row)
 
     def _compute_next_departure(self):
-        cron_next_call = self.env.ref('mass_mailing.ir_cron_mass_mailing_queue').sudo().nextcall
+        cron_job = self.env.ref('mass_mailing.ir_cron_mass_mailing_queue').sudo()
         str2dt = fields.Datetime.from_string
-        cron_time = str2dt(cron_next_call)
+        cron_time = str2dt(cron_job.nextcall)
+        cron_id = cron_job.id
+
+        # A cron.trigger is created when mailing is put "in queue"
+        # The cron workers usually clean up old cron.triggers after finishing their
+        # jobs, i.e older than now(). Thus only most recent than now is relevant
+        now_datetime_buffer = fields.Datetime.now() + relativedelta(minutes=1)
+        cron_triggers = self.env['ir.cron.trigger'].sudo().search([
+            '&',
+            ('cron_id', '=', cron_id),
+            ('call_at', '<', now_datetime_buffer)],
+            order='call_at desc', limit=1)
+        most_recent_trigger_datetime = cron_triggers.call_at
         for mass_mailing in self:
             if mass_mailing.schedule_date:
-                schedule_date = str2dt(mass_mailing.schedule_date)
-                mass_mailing.next_departure = max(schedule_date, cron_time)
+                mass_mailing.next_departure = max(mass_mailing.schedule_date, fields.datetime.now())
```

Comment:
Does not seem this case is tested ?

Reply:
This case kicks in if the client uses the schedule wizard to point to a date in the past (before `now()`).
As of now there is nothing preventing them from doing this, so I don't know if this is a bug or a "feature".
SO if we accept that the user can schedule a date in the past, I probably should add a test for that case!